From: Ahmad Fatoum <a.fatoum@pengutronix.de>
Date: Fri, 3 May 2024 12:15:19 +0200
Subject: [PATCH] mfd: pca9450: don't warn if PCA9450_PWRON_STAT is 0

The register is zeroed on read, so network booting always prints the
warning that the reset reason is unknown.

Skip the warning in this case.

Signed-off-by: Ahmad Fatoum <a.fatoum@pengutronix.de>
Link: https://lore.barebox.org/20240503101519.1226971-1-a.fatoum@pengutronix.de
Signed-off-by: Sascha Hauer <s.hauer@pengutronix.de>
---
 drivers/mfd/pca9450.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/mfd/pca9450.c b/drivers/mfd/pca9450.c
index be357c10239f..89eca9f890e3 100644
--- a/drivers/mfd/pca9450.c
+++ b/drivers/mfd/pca9450.c
@@ -53,6 +53,8 @@ static int pca9450_get_reset_source(struct device *dev, struct regmap *map)
 		break;
 	default:
 		dev_warn(dev, "Unknown reset reason: 0x%02x\n", reg);
+		fallthrough;
+	case 0:
 		type = RESET_UKWN;
 	}
 
